lib/elftools/constants.rb

```diff
@@ -42,120 +42,371 @@ module DF
+      DT_INIT_ARRAYSZ               = 27 # total size of init array
+      DT_FINI_ARRAYSZ               = 28 # total size of fini array
+      DT_RUNPATH                    = 29 # path of libraries for searching
+      DT_FLAGS                      = 30 # flags
+      DT_ENCODING                   = 32 # just a lower bound
+      DT_PREINIT_ARRAY              = 32 # pre-initialization functions array
+      DT_PREINIT_ARRAYSZ            = 33 # pre-initialization functions array size (bytes)
+      DT_SYMTAB_SHNDX               = 34 # address of the +SHT_SYMTAB_SHNDX+ section associated with {DT_SYMTAB} table
+      DT_RELRSZ                     = 35 # :nodoc:
+      DT_RELR                       = 36 # :nodoc:
+      DT_RELRENT                    = 37 # :nodoc:
+
       # Values between {DT_LOOS} and {DT_HIOS} are reserved for operating system-specific semantics.
-      DT_LOOS         = 0x6000000d
-      DT_HIOS         = 0x6ffff000 # see {DT_LOOS}
+      DT_LOOS                       = 0x6000000d
+      DT_HIOS                       = 0x6ffff000 # see {DT_LOOS}
+
       # Values between {DT_VALRNGLO} and {DT_VALRNGHI} use the +d_un.d_val+ field of the dynamic structure.
-      DT_VALRNGLO     = 0x6ffffd00
-      DT_VALRNGHI     = 0x6ffffdff # see {DT_VALRNGLO}
+      DT_VALRNGLO                   = 0x6ffffd00
+      DT_VALRNGHI                   = 0x6ffffdff # see {DT_VALRNGLO}
+
       # Values between {DT_ADDRRNGLO} and {DT_ADDRRNGHI} use the +d_un.d_ptr+ field of the dynamic structure.
-      DT_ADDRRNGLO    = 0x6ffffe00
-      DT_GNU_HASH     = 0x6ffffef5 # the gnu hash
-      DT_ADDRRNGHI    = 0x6ffffeff # see {DT_ADDRRNGLO}
-      DT_RELACOUNT    = 0x6ffffff9 # relative relocation count
-      DT_RELCOUNT     = 0x6ffffffa # relative relocation count
-      DT_FLAGS_1      = 0x6ffffffb # flags
-      DT_VERDEF       = 0x6ffffffc # address of version definition table
-      DT_VERDEFNUM    = 0x6ffffffd # number of entries in {DT_VERDEF}
-      DT_VERNEED      = 0x6ffffffe # address of version dependency table
-      DT_VERSYM       = 0x6ffffff0 # section address of .gnu.version
-      DT_VERNEEDNUM   = 0x6fffffff # number of entries in {DT_VERNEED}
+      DT_ADDRRNGLO                  = 0x6ffffe00
+      DT_GNU_HASH                   = 0x6ffffef5 # the gnu hash
+      DT_TLSDESC_PLT                = 0x6ffffef6 # :nodoc:
+      DT_TLSDESC_GOT                = 0x6ffffef7 # :nodoc:
+      DT_GNU_CONFLICT               = 0x6ffffef8 # :nodoc:
+      DT_GNU_LIBLIST                = 0x6ffffef9 # :nodoc:
+      DT_CONFIG                     = 0x6ffffefa # :nodoc:
+      DT_DEPAUDIT                   = 0x6ffffefb # :nodoc:
+      DT_AUDIT                      = 0x6ffffefc # :nodoc:
+      DT_PLTPAD                     = 0x6ffffefd # :nodoc:
+      DT_MOVETAB                    = 0x6ffffefe # :nodoc:
+      DT_SYMINFO                    = 0x6ffffeff # :nodoc:
+      DT_ADDRRNGHI                  = 0x6ffffeff # see {DT_ADDRRNGLO}
```

Comment:
same here, is it still used?

Reply:
Similar story to `DT_ENCODING`. `DT_ADDRRNGLO` to `DT_ADDRRNGHI` specify a range of values which use the `d_ptr`. This range is inclusive so overlaps with `DT_SYMINFO`.